pylint/checkers/classes.py

```diff
@@ -939,23 +939,22 @@ def _check_unused_private_functions(self, node: astroid.ClassDef) -> None:
                         break
             else:
                 name_stack = []
-                outer_level_names = ""
                 curr = parent_scope
                 # Generate proper names for nested functions
                 while curr != node:
                     name_stack.append(curr.name)
                     curr = curr.parent.scope()
 
-                if name_stack:
-                    outer_level_names = f"{'.'.join(reversed(name_stack))}."
+                outer_level_names = f"{'.'.join(reversed(name_stack))}"
 
-                function_repr = f"{outer_level_names}{function_def.name}({function_def.args.as_string()})"
                 self.add_message(
                     "unused-private-member",
                     node=function_def,
                     args=(
                         node.name,
-                        function_repr,
+                        f"{outer_level_names}.{function_def.name}({function_def.args.as_string()})".lstrip(
+                            "."
+                        ),
```

Comment:
Why did you add the `lstring(".")`? `'.'.join` will never yield a single `.` if that was your intention.

```py
>>> ".".join([])
''
```

Reply:
> Why did you add the `lstring(".")`? `'.'.join` will never yield a single `.` if that was your intention.
> 
> ```python
> >>> ".".join([])
> ''
> ```

This is necessary due to the dot between `outer_level_names` and `function_def.name`

it's true that the join would yield an empty string, but this would result in error messages showing something like

`.Class.my_var` -- therefore i added the lstrip here

Reply:
Makes sense. Thanks for the explanation!